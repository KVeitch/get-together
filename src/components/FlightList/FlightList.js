import React from 'react';
import './FlightList.scss'
import Flight from '../Flight/Flight'

const FlightList = ({ flights }) => {
  const flightList = flights.map((flight) => <Flight flight={flight} key={flight.flightNo} />)

  return ( 
    <section className="section__flights">
      {flightList}
    </section>
  )
}

export default FlightList;
