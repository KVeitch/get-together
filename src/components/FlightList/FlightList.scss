@import '../../variables.scss';

.section__flights {
  border:1px solid $logo-color;
  padding: 10px;
}