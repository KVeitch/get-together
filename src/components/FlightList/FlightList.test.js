/* eslint-disable max-lines-per-function */
import React from 'react';
import { shallow } from 'enzyme';
import FlightList from './FlightList';

describe('FlightList with data', () => {
  const mockFlights = [];
  const mockMath = Object.create(global.Math);
  mockMath.random = () => 0.5;
  global.Math = mockMath;
  const wrapper = shallow(<FlightList flights={mockFlights} />);
  it('should match the snapshot with all the data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('FlightList with data', () => {
  const mockMath = Object.create(global.Math);
  mockMath.random = () => 0.5;
  global.Math = mockMath;
  const mockFlights = [
    {arrival: '01/15/2020', arrivalTime: '1 : 05 PM', depart: '01/12/2020', departTime: '7 : 51 PM', from: 'DEN', to: 'MRY', flights: 'AA 420 and AA 5718'},
    {
      arrival: '01/14/2020',
      arrivalTime: '7 : 36 PM',
      depart: '01/12/2020',
      departTime: '2 : 35 PM',
      from: 'DEN',
      to: 'MRY',
      flights: 'AA 482 and AA 5905'
    },
    {
      arrival: '01/15/2020',
      arrivalTime: '1 : 05 PM',
      depart: '01/12/2020',
      departTime: '5 : 16 PM',
      from: 'DEN',
      to: 'MRY',
      flights: 'AA 2065 and AA 5718'
    },
    {
      arrival: '01/12/2020',
      arrivalTime: '7 : 36 PM',
      depart: '01/12/2020',
      departTime: '5 : 16 PM',
      from: 'DEN',
      to: 'MRY',
      flights: 'AA 2065 and AA 5905'
    },
    {
      arrival: '01/13/2020',
      arrivalTime: '1 : 05 PM',
      depart: '01/12/2020',
      departTime: '2 : 35 PM',
      from: 'DEN',
      to: 'MRY',
      flights: 'AA 482 and AA 5718'
    },
    {
      arrival: '01/12/2020',
      arrivalTime: '11 : 03 PM',
      depart: '01/12/2020',
      departTime: '7 : 51 PM',
      from: 'DEN',
      to: 'MRY',
      flights: 'AA 420 and AA 5837'
    }
  ];
  const wrapper = shallow(<FlightList flights={mockFlights} />);

  it('should match the snapshot with all the data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
