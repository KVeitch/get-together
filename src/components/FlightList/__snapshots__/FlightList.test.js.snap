// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlightList with data should match the snapshot with all the data passed in correctly 1`] = `
<section
  className="section__flights"
>
  Please be patient we're finding flights for you...
</section>
`;

exports[`FlightList with data should match the snapshot with all the data passed in correctly 2`] = `
<section
  className="section__flights"
>
  <Flight
    flight={
      Object {
        "arrival": "01/15/2020",
        "arrivalTime": "1 : 05 PM",
        "depart": "01/12/2020",
        "departTime": "7 : 51 PM",
        "flights": "AA 420 and AA 5718",
        "from": "DEN",
        "to": "MRY",
      }
    }
    key="0.5"
  />
  <Flight
    flight={
      Object {
        "arrival": "01/14/2020",
        "arrivalTime": "7 : 36 PM",
        "depart": "01/12/2020",
        "departTime": "2 : 35 PM",
        "flights": "AA 482 and AA 5905",
        "from": "DEN",
        "to": "MRY",
      }
    }
    key="0.5"
  />
  <Flight
    flight={
      Object {
        "arrival": "01/15/2020",
        "arrivalTime": "1 : 05 PM",
        "depart": "01/12/2020",
        "departTime": "5 : 16 PM",
        "flights": "AA 2065 and AA 5718",
        "from": "DEN",
        "to": "MRY",
      }
    }
    key="0.5"
  />
  <Flight
    flight={
      Object {
        "arrival": "01/12/2020",
        "arrivalTime": "7 : 36 PM",
        "depart": "01/12/2020",
        "departTime": "5 : 16 PM",
        "flights": "AA 2065 and AA 5905",
        "from": "DEN",
        "to": "MRY",
      }
    }
    key="0.5"
  />
  <Flight
    flight={
      Object {
        "arrival": "01/13/2020",
        "arrivalTime": "1 : 05 PM",
        "depart": "01/12/2020",
        "departTime": "2 : 35 PM",
        "flights": "AA 482 and AA 5718",
        "from": "DEN",
        "to": "MRY",
      }
    }
    key="0.5"
  />
  <Flight
    flight={
      Object {
        "arrival": "01/12/2020",
        "arrivalTime": "11 : 03 PM",
        "depart": "01/12/2020",
        "departTime": "7 : 51 PM",
        "flights": "AA 420 and AA 5837",
        "from": "DEN",
        "to": "MRY",
      }
    }
    key="0.5"
  />
</section>
`;
